ring/partition_ring_watcher.go

```diff
@@ -74,25 +78,34 @@ func (w *PartitionRingWatcher) starting(ctx context.Context) error {
 		value = NewPartitionRingDesc()
 	}
 
-	w.updatePartitionRing(value.(*PartitionRingDesc))
+	if err := w.updatePartitionRing(value.(*PartitionRingDesc)); err != nil {
+		return err
+	}
 	return nil
```

Comment:
[nit] Keep it simple, when it can be simple.

```suggestion
	return w.updatePartitionRing(value.(*PartitionRingDesc))
```